src/Service.Tests/Authorization/AuthorizationHelpers.cs

```diff
@@ -63,80 +72,108 @@ public static AuthorizationResolver InitAuthorizationResolver(RuntimeConfig runt
         /// <param name="dbType">Database type configured.</param>
         /// <returns></returns>
         public static RuntimeConfig InitRuntimeConfig(
+            EntitySource entitySource,
             string entityName = TEST_ENTITY,
-            object? entitySource = null,
             string roleName = "Reader",
-            Config.Operation operation = Config.Operation.Create,
+            EntityActionOperation operation = EntityActionOperation.Create,
             HashSet<string>? includedCols = null,
             HashSet<string>? excludedCols = null,
             string? databasePolicy = null,
             string? requestPolicy = null,
             string authProvider = "AppService",
-            DatabaseType dbType = DatabaseType.mssql
+            DatabaseType dbType = DatabaseType.MSSQL
             )
         {
-            Field? fieldsForRole = null;
-
-            if (entitySource is null)
-            {
-                entitySource = TEST_ENTITY;
-            }
+            EntityActionFields? fieldsForRole = null;
 
             if (includedCols is not null || excludedCols is not null)
             {
                 // Only create object for Fields if inc/exc cols is not null.
                 fieldsForRole = new(
-                    include: includedCols,
-                    exclude: excludedCols);
+                    Include: includedCols,
+                    Exclude: excludedCols ?? new());
             }
 
-            Policy policy = new(requestPolicy, databasePolicy);
+            EntityActionPolicy policy = new(requestPolicy, databasePolicy);
 
-            PermissionOperation actionForRole = new(
-                Name: operation,
+            EntityAction actionForRole = new(
+                Action: operation,
                 Fields: fieldsForRole,
                 Policy: policy);
 
-            PermissionSetting permissionForEntity = new(
-                role: roleName,
-                operations: new object[] { JsonSerializer.SerializeToElement(actionForRole) });
+            EntityPermission permissionForEntity = new(
+                Role: roleName,
+                Actions: new EntityAction[] { actionForRole });
 
             Entity sampleEntity = new(
                 Source: entitySource,
-                Rest: null,
-                GraphQL: null,
-                Permissions: new PermissionSetting[] { permissionForEntity },
+                Rest: new(Array.Empty<SupportedHttpVerb>()),
+                GraphQL: new(entityName.Singularize(), entityName.Pluralize()),
```

Comment:
That code path will be hit in the test, as it's used in the constructor of the `RuntimeEntities`. We could probably not set anything here and just leave it to the constructor to do, but it's probably this way because I went through several permutations before finding all the edge cases and the best solution to them.